backend/src/data/migrations/20220818205843_add_admin_group_and_administrator.ts

```diff
@@ -0,0 +1,83 @@
+import { genSalt, hash } from 'bcrypt';
+import { Knex } from 'knex';
+
+const adminCridentials = {
+  email: 'admin@guruhub.club',
+  fullName: 'Admin',
+  password: 'Pa55word',
+};
+
+const ADMIN_GROUP_NAME = 'Admins';
+const USER_PASSWORD_SALT_ROUNDS = 10;
+
+const TableName = {
+  USERS: 'users',
+  PERMISSIONS: 'permissions',
+  USERS_TO_GROUPS: 'users_to_groups',
+  GROUPS: 'groups',
+  GROUPS_TO_PERMISSIONS: 'groups_to_permissions',
+};
+
+async function up(knex: Knex): Promise<void> {
+  const { email, fullName, password } = adminCridentials;
+  const passwordSalt = await genSalt(USER_PASSWORD_SALT_ROUNDS);
+  const passwordHash = await hash(password, passwordSalt);
+
+  const insertedAdmins = await knex(TableName.USERS)
+    .insert({
+      email,
+      fullName,
+      passwordSalt,
+      passwordHash,
+    })
+    .returning('*');
+
+  const [insertedAdmin] = insertedAdmins;
+  const { id: adminId } = insertedAdmin;
+
+  const hasAdminId = Boolean(adminId);
+
+  if (!hasAdminId) {
+    return;
+  }
+
+  const permissionIds = await knex(TableName.PERMISSIONS).select('id');
+  const premissionIdsToGive = permissionIds.map((permission) => permission.id);
+
+  const adminPermissionsDto = {
+    name: ADMIN_GROUP_NAME,
+    permissionIds: premissionIdsToGive,
+    userIds: [adminId],
+  };
```

Comment:
Looks like this object is met only once (and it only wraps `permissionIdsToGive` array that can be used separately), soo it's a little bit useless